### ISSUE TITLE
assign package dependencies to wrong component when package dependencies not installed

### ISSUE THREAD

**Issue by GiladShoham** [teambit/bit#635]
steps to reproduce:
1. create 2 files a and b
2. b -> require('./a')
3. a -> require('lodash')
4. make sure you have package.json with lodash defined
5. make sure lodash is not installed
6. bit show b

bit shows that b is dependent on lodash@^4.17.4  but a is dependent on lodash, not b.

this happens only when the package is not installed

This happens because when we don't found the package, we get it in the missing packages, therefore we lose the relation to the original requirer.
in this case, b -> a -> lodash.
so if the lodash installed, we see it's part of a and we don't relate it to b.
but if not, we will get it as missing in the tree of b because it walk through a.
then we will search for it in the package.json and found a version and add it to the b package deps.

**Comment by jonnysas** [comment]
➤ Gilad Shoham commented:

related to A full explanation is under https://github.com/teambit/bit/issues/690

**Comment by davidfirst** [comment]
@GiladShoham, I wasn't able to reproduce the issue.
When I run 'bit show b', (with or without --json flag), it doesn't show the package-dependencies. 
I made sure that my package.json has lodash as a dependency and node-modules is deleted.
I made sure also that when node-modules is not deleted, I see lodash as a package-dependency when running 'bit show a --json', and not when running 'bit show b --json'.
